prime/src/components/borrow/ManageAccountTransactionButton.tsx

```diff
@@ -218,69 +233,47 @@ export function ManageAccountTransactionButton(props: ManageAccountTransactionBu
+    args: [uniswapPool as Address],
+    chainId: activeChain.id,
+    enabled: enabled,
+  });
+
+  const ante = useMemo(() => {
+    if (!anteData) return GN.zero(18);
+    return GN.fromBigNumber(anteData[0], 18);
+  }, [anteData]);
+
+  const gnAccountEtherBalance = accountEtherBalance ? GN.fromBigNumber(accountEtherBalance.value, 18) : GN.zero(18);
+
   // provide ante if necessary
   const shouldProvideAnte =
     accountEtherBalance &&
-    accountEtherBalance.value.toNumber() < ANTE &&
+    gnAccountEtherBalance.lt(ante) &&
     (accountState.liabilities.amount0.isGtZero() || accountState.liabilities.amount1.isGtZero());
 
-  const isRemovingToken0Collateral = actionIds.some((id, idx) => {
-    if (id === 1) {
-      const actionArg: string = actionArgs[idx];
-      if (!actionArg) return false;
-      const firstArg: string = `0x${actionArg.slice(26, 66).toLowerCase()}`;
-      return firstArg === token0.address;
-    }
-    return false;
-  });
-  const isSwappingToken0ForToken1 = actionIds.some((id, idx) => {
-    if (id === 6) {
-      const actionArg: string = actionArgs[idx];
-      if (!actionArg) return false;
-      const firstArg: string = `0x${actionArg.slice(26, 66).toLowerCase()}`;
-      return firstArg === token0.address;
-    }
-    return false;
-  });
-
-  const isRemovingToken1Collateral = actionIds.some((id, idx) => {
-    if (id === 1) {
-      const actionArg: string = actionArgs[idx];
-      if (!actionArg) return false;
-      const firstArg: string = `0x${actionArg.slice(26, 66).toLowerCase()}`;
-      return firstArg === token1.address;
-    }
-    return false;
-  });
-  const isSwappingToken1ForToken0 = actionIds.some((id, idx) => {
-    if (id === 6) {
-      const actionArg: string = actionArgs[idx];
-      if (!actionArg) return false;
-      const firstArg: string = `0x${actionArg.slice(26, 66).toLowerCase()}`;
-      return firstArg === token1.address;
-    }
-    return false;
-  });
-
   const calldata = canConstructTransaction
     ? ethers.utils.defaultAbiCoder.encode(['uint8[]', 'bytes[]', 'uint144'], [actionIds, actionArgs, positions])
     : null;
 
   const { config: contractConfig } = usePrepareContractWrite({
     address: accountAddress,
-    abi: MarginAccountAbi,
+    abi: borrowerABI,
     functionName: 'modify',
     chainId: activeChain.id,
-    args: [
-      ALOE_II_FRONTEND_MANAGER_ADDRESS[activeChain.id],
-      calldata,
-      [
-        isRemovingToken0Collateral || isSwappingToken0ForToken1,
-        isRemovingToken1Collateral || isSwappingToken1ForToken0,
-      ],
-    ],
-    overrides: { value: shouldProvideAnte ? ANTE + 1 : undefined },
-    enabled: canConstructTransaction && enabled && !transactionWillFail && !needsApproval[0] && !needsApproval[1],
+    args: [ALOE_II_FRONTEND_MANAGER_ADDRESS[activeChain.id], calldata as `0x${string}`, oracleSeed ?? Q32],
+    overrides: { value: shouldProvideAnte ? ante.recklessAdd(1).toBigNumber() : undefined },
```

Comment:
Don't need to recklessAdd(1) anymore.

Reply:
Punting, remind me when this change takes effect : )